### PR TITLE
Add JWST data

### PR DESCRIPTION
**Pull request by mhuertascompany**
Adding scripts to create JWST imaging datasets

### CODE REVIEW

File: scripts/jwst/.DS_Store
Comment (EiffL) [inline scripts/jwst/.DS_Store:1]: Please remove

File: scripts/jwst/build_parent_sample.py [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+
+                # Initialize the example with image data
+            #print(np.zeros((image.shape[1],image.shape[2])).shape)
+             # Automatically create _filters by formatting each entry in _filter_list (astropile nomenclature)
+            filters = [f'jwst_nircam_{filter_name}' for filter_name in filter_list]
+            images.append({
+                        'object_id': id,
+                        'image_band': np.array([f.lower().encode("utf-8") for f in filters], dtype=_utf8_filter_type),
+                        'image_array': image,
+                        'image_psf_fwhm': image,
```
Comment (EiffL) [inline scripts/jwst/build_parent_sample.py:364]: Can you compute the FWHM somehow? For space based it's fine to have a single number

Comment (EiffL) [reply scripts/jwst/build_parent_sample.py:364]: And you can add another entry for 'image_psf_array'

File: scripts/jwst/build_parent_sample.py [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+
+            key = str(id)
+
+            
+
+            # Get the smallest shape among all images
+            stamp = JWST_multilambda[f][c]
+            s_x=stamp.shape[0]
+            s_y=stamp.shape[1]
+            #s_x = min([stamp.shape[0] for f in filter_list])
```
Comment (EiffL) [inline scripts/jwst/build_parent_sample.py:318]: can you remove the commented text?

File: scripts/jwst/build_parent_sample.py [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+       
+
+
+    # Loop over the indices and yield the requested data
+    #for c, id, ra, dec in zip(range(len(idvec)),idvec, ravec,decvec):
+    if not os.path.exists(group_filename):
+        for row in group:
+            c = row['index']  # Assuming there is an 'index' column specifying the object's index
+            id = row['object_id']  # Assuming there is an 'id' column
+            #ra = row['ra']  # Assuming there is a 'ra' column
```
Comment (EiffL) [inline scripts/jwst/build_parent_sample.py:307]: where is my ra and dec?

Comment (EiffL) [reply scripts/jwst/build_parent_sample.py:307]: Ok good, maybe remove the commented code :-)

File: scripts/jwst/jwst.py [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        """
+        float_features = self.config.float_features if self.config.float_features is not None else []
+
+
+        # Starting with all features common to image datasets
+        features = {
+            'image': Sequence(feature={
+                'band': Value('string'),
+                'array': Array2D(shape=(self._image_size, self._image_size), dtype='float32'),
+                'psf_fwhm': Array2D(shape=(self._image_size, self._image_size), dtype='float32'),
```
Comment (EiffL) [inline scripts/jwst/jwst.py:270]: This should be updated to return a scalar number

File: scripts/jwst/jwst.py [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+                    #print(keys)
+                # Preparing an index for fast searching through the catalog
+                sort_index = np.argsort(data["object_id"])
+                sorted_ids = data["object_id"][:][sort_index]
+
+                for k in keys:
+                    # Extract the indices of requested ids in the catalog 
+                    i = sort_index[np.searchsorted(sorted_ids, k)]
+                    #print('index',i,j)
+                    #print(data['image_band'][i][j].decode('utf-8'))
```
Comment (EiffL) [inline scripts/jwst/jwst.py:335]: remove commented code

File: scripts/jwst/jwst_old.py
Comment (EiffL) [inline scripts/jwst/jwst_old.py:1]: to be removed

**Review by EiffL — CHANGES_REQUESTED** [review]
This looks very good! could you just remove some of the experimental code that you had and commented text just to make it cleaner.

**Comment by LTMeyer** [comment]
Tests are now passing. The issue was related to how tables were obtained and multiple downloads and dataset processing in the same folder. The `cosmos` field was also problematic. @mhuertascompany solved these problems.

I removed the commented or unused code, formatted it with ruff and sorted the imports.
I don't know if @EiffL 's remaining comments have been addressed (for returning a scalar number and adding another entry for `image_psf_array`).

**Review by EiffL — APPROVED** [review]
I made some small modifications, and I think it's all good to merge.

**Pull request merged**